File: src/UI/TradeTrackerUI/src/app/models/transactions/paged-transactions-with-links.ts
import { Link } from "../shared/link";
import { PaginationMetadata } from "../shared/pagination-metadata";
import { TransactionWithLinks } from "./transaction-with-links";

export interface PagedTransactionsWithLinks {
    metadata: PaginationMetadata;
    items: TransactionWithLinks[];
    links: Link[];
}

File: src/Core/TradeTracker.Application/Features/Transactions/Queries/GetTransactions/PagedTransactionsWithLinks.cs
using System.Collections.Generic;
using TradeTracker.Application.Common.Models.Resources.Responses;
using TradeTracker.Application.Models.Common.Resources.Responses;

namespace TradeTracker.Application.Features.Transactions.Queries.GetTransactions
{
    public class PagedTransactionsWithLinks 
    {
        public PaginationMetadata Metadata { get; set; }

        public IEnumerable<TransactionForReturnWithLinks> Items { get; set; }

        public IEnumerable<Link> Links { get; set; }
    }
}